### ISSUE TITLE
Create a proper locale folder for language translation support.

### ISSUE THREAD

**Issue by Oarcinae** [Oarcinae/FactorioScenarioMultiplayerSpawn#71]
Would be nice to support other languages.

**Comment by Oarcinae** [comment]
@BurtGummer I added support. I haven't finished everything but let me know if you have any German translations and I can add it in (or you can pull request if you want to.)

**Comment by Oarcinae** [comment]
Depending on much people want this, I will then go in and do the other GUI text remaining. Otherwise if nobody uses it, it's not worth the effort.

**Comment by BurtGummer** [comment]
@Oarcinae i have almost complete the translation. One thing i dont understand:

oarc-set-respawn-loc-cooldown=Set Respawn Cooldown Remaining: __1__

What's that for? If a player die?

**Comment by Oarcinae** [comment]
So under the Spawn Ctrl button, there are a couple of controls:
![image](https://user-images.githubusercontent.com/23324249/55000339-b1f16c80-4fa8-11e9-9bfc-e566557a4567.png)

1. You can allow people to join your base.
2. You can change your respawn point for when you die. If you set a new respawn location, this means when you die, you will come back in the new location that you set.

The intended use is that after you leave your starting base, you might build a new base, or join someone else. If you die, you will spawn back at your old starting base, but this option allows you to change that.

**Comment by Oarcinae** [comment]
Also, there are parts of the GUI text that I generate by connecting several phrases together like this:

```
oarc-buddy-txt-main-team=the main team
oarc-buddy-txt-new-teams=on separate teams
oarc-buddy-txt-buddy-team=a buddy team
oarc-buddy-txt-moat= surrounded by a moat 
oarc-buddy-txt-near=near to the center of the map!
oarc-buddy-txt-far=far from the center of the map!
oarc-buddy-txt-would-like= would like to join 
oarc-buddy-txt-next-to-you= next to you
```

I'm not sure how well that will translate because it is combined together to form one big sentence. It might be hard to get the grammar/translation correct for that one but I'm not sure of a good solution.

The end result is something like "Player A is requesting to join the main team surrounded by a moat next to you near to the center of the map!". This prompt is displayed to the player when creating the buddy spawn depending on what options are selected.

**Comment by BurtGummer** [comment]
OK, i think i have to play a bit with the translation to see it in action.

**Comment by Oarcinae** [comment]
@BurtGummer If you have a file that is "good enough" to use. I can add that for now and we can tweak it later. I might just use google translate for now and then let people complain about weird translation issues.

**Comment by BurtGummer** [comment]
@Oarcinae i have create my first PR, hope it works. And a question, how can i switch to a other language? There are no option in example-config.lua?

**Comment by IIPoliII** [comment]
@BurtGummer to swtich to other languages switch in the game your language but i think he needs to modify something somewhere too in the config

**Comment by IIPoliII** [comment]
I also did my first PR for french ^^

**Comment by Oarcinae** [comment]
@BurtGummer Yes, this is correct, the only way to change language is in the game settings. This is unrelated to the scenario. As long as the scenario has the correct /locale/language folder, then it should work AFTER restarting factorio.

**Comment by Oarcinae** [comment]
@BurtGummer @IIPoliII 
Thanks for the PRs! I accepted both, if you need to make further changes feel free to submit more in the future. I still need to add more text translations for the other GUIs like Info panel and other things. I will let you know when I get to that.

**Comment by davv1** [comment]
I can make Polish in free time @Oarcinae

**Comment by Oarcinae** [comment]
@davv1 Sounds good, just take a look at the /locale folder and copy it. You can post the file here, or a link, or do a git pull request. Whatever you want to do, I appreciate the help! thanks!
